### DISCUSSION TITLE
Status bar diagnostic button doesn't close open diagnostics tab.

### DISCUSSION THREAD

**Discussion by Angelk90** [zed-industries/zed#23921] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

https://github.com/zed-industries/zed/assets/20476002/72e3a2b1-63d5-4359-bb03-46aeae05f71e

### Environment

```
Zed: v0.139.3 (Zed Preview)
OS: macOS 14.1.1
Memory: 16 GiB
Architecture: x86_64
```

**Comment by JosephTLyons** [comment]
This is expected, since the diagnostics multi-buffer is just a tab, it is treated mostly the same as any other. The diagnostics tab can be opened in multiple panes if you want (split), so right now, that button is just an open-only button.

**Comment by Angelk90** [comment]
@JosephTLyons : 
Why not put it here then?
<img width="221" alt="Screenshot 2024-06-12 alle 11 56 01" src="https://github.com/zed-industries/zed/assets/20476002/488f74a4-6652-4a91-9c5f-c7c922bee1b8">


If I see that button I expect it can both open and close it.

How does terminal panel.

**Comment by notpeter** [comment]
@iamnbutler Do you have any thoughts on this? Every other icon in our status bar, if you click on it once it shows the panel and if you click on it again it hides.  Is this something that could fixed this Quality week or is it operating as designed?

**Comment by Angelk90** [comment]
@notpeter : Does it work for you? If you click when it's open, does it close?

**Comment by notpeter** [comment]
No. Clicking on the icon does not hide/close for me.
This behavior deviates from all the other status bar icons.

**Comment by iamnbutler** [comment]
We know it works this way – the issue is diagnostics is not singular. Multiple can be open, and clicking the icon focuses any non-foregrounded diagnostic tab.

Essentially this would require changing the way diagnostics works to make it singular.

I'm not sure if that is a change we would be willing to make.
